### PR TITLE
DOC: Update of the 'getting started' pages in the sphinx section of the documentation

### LINKED ISSUE #26831: DOC: Discussion on new getting started page (Numfocus Grant)
As currently prepared together with @jorisvandenbossche, the following document is a proposal on how to adjust the getting started page and section of the documentation: https://docs.google.com/document/d/1Rc_eql5KLrdf0c582KyWfs2ADVNxbJy4jfosnqdrVak/edit?usp=sharing

The general idea is to split the current 10 minutes into 10 x 1' topics of Pandas. Input is very welcome on:
- the topics to pick and how to introduce them
- the content of the introduction tutorial linked to each of the topics

### PR DESCRIPTION
**Pull request by stijnvanhoey**
- [ ] closes #26831
 
This PR provides an update of the getting started pages of the Pandas documentation, following from the discussion in #26831 and the [proposal](https://docs.google.com/document/d/1Rc_eql5KLrdf0c582KyWfs2ADVNxbJy4jfosnqdrVak/edit#heading=h.mqz2f6gbl3sd). The update went together with the creation of the new theme and tries to integrate as much as possible with the new layout (using bootstrap elements).  This PR focuses on the getting started section and adds new sections to the documentation:

 - An update of the 'getting started' intro page of the sphinx documentation to provide new users some more guidance on the 'first steps' with Pandas: Directions for installation, a short intro on main Pandas features each linking to a dedicated introduction tutorial (see next point), specific info targeted to people with another background and reference to more tutorials.
- A set of tutorials which introduce some key features of the Pandas package. Each tutorial is setup as a series of 'tasks/questions' so users get a first idea of the type of problems they can solve with Pandas for that topic. All tutorials provide a set of [returning elements](https://stijnvanhoey.github.io/pandas-getting-started-tutorials/tutorial-elements/). Both the intros as well as the tutorials itself try to guide people as much as possible to related sections of the user guide. 
- A set of schemas to illustrate certain concepts, with according to the new theme. These schemas can be used in other locations in the user guide as well.

The aim is to move the [dsintro](https://pandas.io/docs/getting_started/dsintro.html) and the [basics](https://pandas.io/docs/getting_started/basics.html) to the general user guide, as both are too extensive for a getting started section. The 10 intro tutorials try to provide an alternative that is more fit to people starting with Pandas. 

There is still some work to do (typo's in the tutorials, additional schemas...), but input is certainly welcome and appreciated, I'll continue improving it. See https://stijnvanhoey.github.io/example-pandas-docs/getting-started/getting_started/index.html for a live preview.

Note: (1) a separate PR is prepared for the general sphinx intro page with @jorisvandenbossche, see #31148  (2) some of the layout issues are more general and related to the [new sphinx theme](https://github.com/pandas-dev/pydata-bootstrap-sphinx-theme) and will be tackled there (e.g. spacing around titles and subtitles)

Some pictures:
- The intro to the 10 tutorials:
image

- the coming from section :
image

- example of 'remember' section in each tutorial:
image

### CODE REVIEW

**Comment by jreback** [comment]
is there a rendered link to this?

**Comment by stijnvanhoey** [comment]
@jreback working on it, currently building

**Comment by jreback** [comment]
love the scrolling highlite on the rhs :->

File: doc/source/index.rst.template [hunk 1]
```diff
@@ -1,55 +1,31 @@
...
+programming language.
 
-`Binary Installers <https://pypi.org/project/pandas>`__ |
-`Source Repository <https://github.com/pandas-dev/pandas>`__ |
+See the :ref:`overview` for more detail about what's in the library.
 
 {% if single_doc and single_doc.endswith('.rst') -%}
 .. toctree::
     :maxdepth: 3
-    :titlesonly:
```
Comment (jorisvandenbossche) [inline doc/source/index.rst.template]: @stijnvanhoey this part might still be needed to get the correct toctree? (so it doesn't use the section headers on the getting_started/index.rst page)

Comment (stijnvanhoey) [reply doc/source/index.rst.template]: Correct! I did a checkout to a wrong version to provide the input for your PR.

**Comment by jorisvandenbossche** [comment]
@stijnvanhoey the doc build is required to pass without any warning or error.

First, there are some warnings about images:

```
/home/runner/work/pandas/pandas/doc/source/getting_started/index.rst:: WARNING: image file not readable: getting_started/../../_static/schemas/01_table_dataframe.svg
/home/runner/work/pandas/pandas/doc/source/getting_started/index.rst:: WARNING: image file not readable: getting_started/../../_static/schemas/02_io_readwrite.svg
/home/runner/work/pandas/pandas/doc/source/getting_started/index.rst:: WARNING: image file not readable: getting_started/../../_static/schemas/03_subset_columns_rows.svg
/home/runner/work/pandas/pandas/doc/source/getting_started/index.rst:: WARNING: image file not readable: getting_started/../../_static/schemas/04_plot_overview.svg
/home/runner/work/pandas/pandas/doc/source/getting_started/index.rst:: WARNING: image file not readable: getting_started/../../_static/schemas/05_newcolumn_2.svg
/home/runner/work/pandas/pandas/doc/source/getting_started/index.rst:: WARNING: image file not readable: getting_started/../../_static/schemas/06_groupby.svg
/home/runner/work/pandas/pandas/doc/source/getting_started/index.rst:: WARNING: image file not readable: getting_started/../../_static/schemas/07_melt.svg
/home/runner/work/pandas/pandas/doc/source/getting_started/index.rst:: WARNING: image file not readable: getting_started/../../_static/schemas/08_concat_row.svg
```

I suppose this are the images in the accordion on gettings_started/index.rst. And I think they need one level of `/../` less compared to the ones in the actual tutorials? (but strange that it works in the preview https://stijnvanhoey.github.io/example-pandas-docs/getting-started/getting_started/index.html ..)

Then there are also some warnings generated by the examples in the timeseries tutorial:

```
>>>-------------------------------------------------------------------------
Warning in /home/runner/work/pandas/pandas/doc/source/getting_started/intro_tutorials/09_timeseries.rst at block ending on line 279
Specify :okwarning: as an option in the ipython:: block to suppress this message
----------------------------------------------------------------------------
/home/runner/work/pandas/pandas/pandas/core/arrays/datetimes.py:1099: UserWarning: Converting to PeriodArray/Index representation will drop timezone information.
  UserWarning,
<<<-------------------------------------------------------------------------


>>>-------------------------------------------------------------------------
Warning in /home/runner/work/pandas/pandas/doc/source/getting_started/intro_tutorials/09_timeseries.rst at block ending on line 356
Specify :okwarning: as an option in the ipython:: block to suppress this message
----------------------------------------------------------------------------
/home/runner/work/pandas/pandas/pandas/core/arrays/datetimes.py:1099: UserWarning: Converting to PeriodArray/Index representation will drop timezone information.
  UserWarning,
<<<-------------------------------------------------------------------------
```

the first one is coming from 

```
.. ipython:: python

    @savefig 09_time_section.png
    no_2["2019-05-20":"2019-05-21"].plot();
```

That's a strange warning, but not something you need to be concerned about in this PR. To get CI green here you can add `:okwarning:` to that ipython directive (and the second one that generates the warning), like:

```
.. ipython:: python
    :okwarning:

    @savefig 09_time_section.png
    no_2["2019-05-20":"2019-05-21"].plot();
```

**Comment by jorisvandenbossche** [comment]
Opened https://github.com/pandas-dev/pandas/issues/31205 for the warning

**Comment by WillAyd** [comment]
This looks great!

File: doc/source/getting_started/intro_tutorials/04_plotting.rst [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+{{ header }}
+
+.. ipython:: python
+
+    import pandas as pd
+    import matplotlib.pyplot as plt
+    # use the pandas main colors
+    import matplotlib as mpl
+    mpl.rcParams['axes.prop_cycle'] = mpl.cycler(color=["#150458", "#FFCA00", "#E70488"])
```
[the remark this answers is not in the document]
Comment (stijnvanhoey) [reply doc/source/getting_started/intro_tutorials/04_plotting.rst]: done

**Comment by jorisvandenbossche** [comment]
Question on how to move forward here. 
Are people fine with merging this mostly as is? (I am still reviewing a few details) The content already got some review on the separate repo (https://github.com/stijnvanhoey/pandas-getting-started-tutorials/)

**Review by TomAugspurger — APPROVED** [review]

**Comment by WillAyd** [comment]
Merge away - I think to 1.0.2 right?

**Comment by TomAugspurger** [comment]
Yep. Thanks @stijnvanhoey!

**Pull request merged**